### ISSUE TITLE
String-form generic parameter in base class list not recognized with PEP 695 syntax 

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#7737]
Pyright generates a false error on following code:

```python
class MyList[T](list["T"]):
                   # ^^^ "T" is not defined
  pass
```
While works fine for similar code:
```python
class MyList[T](list[T]):
    pass

class MyType[T]:
    value: list["T"]

T = TypeVar("T")
class MyList(Generic[T], list["T"]):
    pass
```

version: Pylance vscode extension, v2024.4.1

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.360, which I just published.
